src/modules/mavlink/mavlink_receiver.cpp

```diff
@@ -407,8 +422,23 @@ MavlinkReceiver::handle_message_command_int(mavlink_message_t *msg)
 	vcmd.param2 = cmd_mavlink.param2;
 	vcmd.param3 = cmd_mavlink.param3;
 	vcmd.param4 = cmd_mavlink.param4;
-	vcmd.param5 = ((double)cmd_mavlink.x) / 1e7;
-	vcmd.param6 = ((double)cmd_mavlink.y) / 1e7;
+
+	if (cmd_mavlink.x == INT32_MAX && cmd_mavlink.y == INT32_MAX) {
+		// INT32_MAX for x and y likely means to ignore it.
+		vcmd.param5 = (double)NAN;
+		vcmd.param6 = (double)NAN;
+
+	} else if (cmd_mavlink.x == 0x7ff80000 && cmd_mavlink.y == 0x7ff80000) {
```

Comment:
I'm a bit on the fence here: these values could very well be intended.
As it's a bug on the other side, I'd rather just print a warning, but keep the value.

Reply:
@bkueng I agree that it could happen for one of the fields but for x and y at the same time? Probably never ever...

Reply:
We're talking about command inputs to a flight controller. I'd rather have certainty and guarantees here, even if it is indeed very unlikely.

Reply:
So you rather risk flyaways rather than passing a command along with NAN which will then get handled properly later, and most likely result in the command not being executed.

Reply:
These numbers would happen if:
1. We are sending lat and lon set to 214.695936 degrees which doesn't make any sense in the first place because latitude only goes from -90 to 90.
2. We are sending x/y in meters to go to 214695.936 meters in both directions, so ~ 214 km, I guess that could in theory happen.

Reply:
If you use the wrong commands/don't follow the spec, then yes, you have to expect undesired results. That can hardly be surprising. Once you add such logic, it's almost impossible to remove/cleanup later, as someone might depend on it. And to be clear I have nothing against API hardening.
But if you feel strong about it, go ahead.

Reply:
That's a very important point I didn't think of.

Reply:
Given it complains with an error, I would say you should not start to depend on it.

Maybe better would be NACK it with invalid argument. That would make sure it doesn't go unnoticed, or have unexpected side effects when going 214km both ways.

Reply:
@bkueng I pushed another version. Let me know if that would work for you.